### ISSUE TITLE
PAML run method docstrings - return value

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#1708]
The docstrings for the PAML run methods seem a little confusing/out-of-date about the return values.

https://github.com/biopython/biopython/blob/biopython-172/Bio/Phylo/PAML/_paml.py#L88
*"Return a process signal so the user can determine if the execution was successful (return code 0 is successful, -N indicates a failure)."*
Actually returns None, but will raise an exception if the return code was non-zero.

https://github.com/biopython/biopython/blob/biopython-172/Bio/Phylo/PAML/codeml.py#L172
https://github.com/biopython/biopython/blob/biopython-172/Bio/Phylo/PAML/baseml.py#L160
*"Return a process signal so the user can determine if the execution was successful (return code 0 is successful, -N indicates a failure)."*
Actually returns the parsed data, or None if parse=False. Again, should raise an exception if the return code was non-zero.

https://github.com/biopython/biopython/blob/biopython-172/Bio/Phylo/PAML/yn00.py#L97
Missing a docstring, but behaves as above.

Could you look at this please @brandoninvergo - even just to confirm my reading of the code, then we can tag this as an easy-fix issue for a potential new contributor to tackle.

**Comment by brandoninvergo** [comment]
Yes, it looks like your reading is correct to me. The doc strings in general look like they could be cleaned up and improved (i.e. `read()` doesn't say what it returns at all).

I'm in end-of-postdoc-looking-for-a-job mode so it would be better to suggest this as low-hanging-fruit, I think.

**Comment by peterjc** [comment]
Thanks - let's see if we have any volunteers.

**Comment by funcylambda** [comment]
I'll pick this up :)

**Comment by farAtGitHub** [comment]
The last PR is over a year old and still open. Is there anything I can do?

**Comment by peterjc** [comment]
I think it would be reasonable to start from the work on #1824 (preserving the original commits and authorship if you are comfortably enough with git to to that), and addressing the feedback which stopped it being merged then.

**Comment by farAtGitHub** [comment]
Problem is: Now I cannot commit because of one
`D100 Missing docstring in public module`
and a ton of
`Q000 Remove bad quotes`
per changed file (baseml.py and codeml.py)
and one 
`Bio/Phylo/PAML/codeml.py:129:80: E501 line too long (83 > 79 characters)`
due to 'heavy' indenting.

**Comment by farAtGitHub** [comment]
I fixed the D209 issues in https://github.com/biopython/biopython/pull/2489 building on #1824's work.

**Comment by peterjc** [comment]
Strange that you are seeing E501, our ``.flake8`` setting should be ignoring that - we'll likely adopt 88 as part of the black style though:

https://github.com/biopython/biopython/blob/master/.flake8

**Comment by peterjc** [comment]
For fixing the quotes automatically, I suggest:

```
pip install unify
unify --quote \" --in-place Bio/Phylo/PAML/*.py
```

**Comment by peterjc** [comment]
Or, since we've allready applied black to the PAML folder, this would fix the quotes and more:

```
pip install black
black Bio/Phylo/PAML/*.py
```

**Comment by farAtGitHub** [comment]
Perhaps I need to do some more work on my setup. At least the commit hook is doing fine.
I'll try your suggestions.
One thing before I destroy everything: Double quotes are the bad guys, we want single quotes, right?

**Comment by peterjc** [comment]
No, we're following black and using double quotes.

**Comment by farAtGitHub** [comment]
So I merged a (more) current version of master into the old (+ mine) changes last night, solved  resulting conflicts and ran the tests. All flake8 issues are gone, too. :-)
Of course master has changed again already but I think #2489 is ready for review now.

**Comment by vishesh2802** [comment]
Hello community, I am vishesh 2nd-year undergraduate I would like to start my contribution in the repo, Saw this issue open currently, Can someone guide me on where to contribute to this issue, if not then What could be a good first issue for me.

**Comment by peterjc** [comment]
This issue (and any with the "good first issue" tag) should be fine to tackle as a beginner, but you will need to explore the command line phylogenetic tool PAML to understand the context.

**Comment by RUCHI21CODES** [comment]
Sir, I want to contribute in this project, can you please assign me the issue and could you please tell me briefly what am I supposed to do this because the last update on the issue was done long ago

**Comment by peterjc** [comment]
@RUCHI21CODES no one else seems to be working on this, so please have a go. You will need to read the work in #1824 and #2489 and importantly the feedback on those pull requests.

You will need some background knowledge of the phylogenetic tool PAML, and at very least to try running it at the command line yourself.

**Comment by peterjc** [comment]
Closed by #5034 - thank you all!
